tinygrad/runtime/ops_metal.py

```diff
@@ -10,7 +10,9 @@
 from tinygrad.shape.symbolic import Variable, Node
 
 class MetalAllocator(LRUAllocator):
-  def _do_alloc(self, size, dtype, device, **kwargs): return METAL.device.newBufferWithLength_options_(size*dtype.itemsize, Metal.MTLResourceStorageModeShared)
+  def _do_alloc(self, size, dtype, device, **kwargs):
+    assert (buf_len := size*dtype.itemsize) < (max_len := METAL.device.maxBufferLength()) and (buf := METAL.device.newBufferWithLength_options_(buf_len, Metal.MTLResourceStorageModeShared)), f"Failed to create a {buf_len/1e9:5.2f} GB Metal buffer. {f'The max buffer length is {max_len/1e9:5.2f} GB.' if buf_len > max_len else f'Metal returned {buf}.'}"
```

Comment:
Please use two asserts!

Reply:
also don't use `:=` in assert, assert should not have side effects

Reply:
Are they okay for use within the assert messages? I see one example in helpers.py:30.

Reply:
i think if it's just for the assertion message and it reads better it's fine. here it's used outside of assertion